overwatch/src/services/mod.rs

```diff
@@ -23,9 +23,9 @@ pub trait ServiceData {
     /// Service settings object
     type Settings;
     /// Service state object
-    type State;
+    type State: ServiceState<Settings = Self::Settings>;
     /// State operator
-    type StateOperator;
+    type StateOperator: StateOperator<State = Self::State>;
```

Comment:
We don't really need the bounds. Lets go with the bounds free option. States are tied up in trait impl.